test/fixtures/config.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const Config = require('../../')
+
+const LEVELS = [
+  'fatal'
+, 'error'
+, 'warn'
+, 'info'
+, 'debug'
+, 'trace'
+, 'silent'
+]
+
+const input = [
+  Config
+    .string('rsshosts')
+    .default('localhost')
+    .required()
+    .desc('The host or hosts string for RSS')
+, Config
+    .number('rssport')
+    .default(6371)
+    .desc('The port for RSS')
+, Config
+    .number('port')
+    .default(3000)
+    .min(1025)
+    .desc('The port on which this service runs')
+, Config
+    .enum('loglevel')
+    .default('info')
+    .desc('The loglevel for the service')
+    .values(LEVELS)
+, Config
+    .regex('bisquick')
+    .match('biscuits')
+, Config
+    .boolean('pretty-print')
+    .default(false)
+    .desc('When false, JSON is logged. When true, the logs are prettier.')
+, Config
+    .number('ifr-active-ldbe')
+    .default(2)
+    .desc('I have no idea')
+, Config
+    .number('ifr-deferred-ldbe')
+    .default(1)
+    .desc('I have no idea')
```

Comment:
nit: some of these examples could be better and/or more generic

Reply:
agreed

Reply:
I re-worked this entire file to be generic and only test each `type` once.  I also added document sections for `.enum` which appeared to have been missing before.